src/main/java/lotto/LottoTicketGenerator.java

```diff
@@ -15,22 +16,32 @@ public LottoTicketGenerator(LottoTicketBall lottoTicketBall) {
         this.lottoTicketBall = lottoTicketBall;
     }
 
-    public List<LottoTicket> purchaseLotto(int purchaseAmount) {
+    public LottoTicketCollection purchaseLotto(int purchaseAmount, List<List<Integer>> numbers) {
         if (purchaseAmount < PRICE_PER_TICKET) {
             throw new IllegalArgumentException("로또를 구매할 수 없습니다.");
         }
-        return generateTickets(calculateLottoCount(purchaseAmount));
+
+        return new LottoTicketCollection(generateAutoTickets(calculateLottoCount(purchaseAmount)),
+            generateManualTickets(numbers));
```

Comment:
```suggestion
        List<LottoTicket> manual = generateManualTickets(numbers);
        List<LottoTicket> auto = generateAutoTickets(calculateLottoCount(purchaseAmount) - manual.size());
        return new LottoTicketCollection(auto, manual);
```
수동 수 만큼 자동에서 제거해줘야 하지 않을까요?